server/service/integration_core_test.go

```diff
@@ -4119,6 +4119,25 @@ func (s *integrationTestSuite) TestAppConfig() {
   }`), http.StatusOK, &acResp)
 	assert.Equal(t, "test", acResp.OrgInfo.OrgName)
 
+	// the global agent options were not modified by the last call, so the
+	// corresponding activity should not have been created.
+	var listActivities listActivitiesResponse
+	s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &listActivities, "order_key", "id", "order_direction", "desc")
+	if !assert.Len(t, listActivities.Activities, 0) {
+		// if there is an activity, make sure it is not edited_agent_options
+		require.NotEqual(t, fleet.ActivityTypeEditedAgentOptions, listActivities.Activities[0].Type)
+	}
+
+	// test a change that does modify the agent options.
+	s.DoJSON("PATCH", "/api/latest/fleet/config", json.RawMessage(`{
+    "agent_options": { "foo": 1 }
+  }`), http.StatusOK, &acResp)
+	s.DoJSON("GET", "/api/latest/fleet/activities", nil, http.StatusOK, &listActivities, "order_key", "id", "order_direction", "desc")
+	require.True(t, len(listActivities.Activities) > 0)
+	require.Equal(t, fleet.ActivityTypeEditedAgentOptions, listActivities.Activities[0].Type)
+	require.NotNil(t, listActivities.Activities[0].Details)
```

Comment:
here and below: is there a chance of another test running in parallel and creating more activities, thus making this fail? I have used this in the past, but maybe it was overkill?

https://github.com/fleetdm/fleet/blob/fe9cdb7017f7718a136243c81ec5b2c6ef70fb9c/server/service/integration_enterprise_test.go#L1235-L1242

Reply:
Integration tests run in their own dedicated test database, so unless we start making subtests of a given integration test suite run in parallel (which we shouldn't), this should not be an issue.